### PR TITLE
feat(versions): add versions.json to describe all versions

### PR DESCRIPTION
**Pull request by lahabana**
This enables to use links contextual to the release and to always
have up to date links to release artifacts!

Signed-off-by: Charly Molter <charly.molter@konghq.com>

### CODE REVIEW

File: docs/docs/dev/installation/openshift.md [hunk 1]
```diff
@@ -26,11 +26,11 @@ curl -L https://kuma.io/installer.sh | sh -
 
 You can also download the distribution manually. Download a distribution for the **client host** from where you will be executing the commands to access OpenShift:
 
-* [CentOS](https://download.konghq.com/mesh-alpine/kuma-1.4.1-centos-amd64.tar.gz)
-* [RedHat](https://download.konghq.com/mesh-alpine/kuma-1.4.1-rhel-amd64.tar.gz)
-* [Debian](https://download.konghq.com/mesh-alpine/kuma-1.4.1-debian-amd64.tar.gz)
-* [Ubuntu](https://download.konghq.com/mesh-alpine/kuma-1.4.1-ubuntu-amd64.tar.gz)
-* [macOS](https://download.konghq.com/mesh-alpine/kuma-1.4.1-darwin-amd64.tar.gz) or `brew install kumactl`
+* <a :href="'https://download.konghq.com/mesh-alpine/kuma-' + $page.latestVersion + '-centos-amd64.tar.gz'">CentOS</a>
```
Comment (michaelbeaumont) [inline docs/docs/dev/installation/openshift.md:29]: shame this doesn't work with markdown..

File: docs/docs/dev/versions.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "helm": ["0.8.0","0.8.1"],
+  "kuma": ["1.5.0-rc2"]
```
Comment (michaelbeaumont) [inline docs/docs/dev/versions.json:3]: should this be here?

Comment (lahabana) [reply docs/docs/dev/versions.json:3]: Yeah it's to have a version in the `dev` branch

File: docs/docs/dev/installation/openshift.md (hunk 1 above)
Comment (lahabana) [reply docs/docs/dev/installation/openshift.md:29]: Yeah depressing! Took me hours to try to find a way to make it work

File: docs/.vuepress/versions.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    return all[all.length - 1];
+  },
+  /// All X.Y.x versions
+  allMinors: () => {
+    let out = [];
+    for (let key in _versions) {
+      out.push(key);
+    }
+    return out;
+  },
```
Comment (bartsmykla) [inline docs/.vuepress/versions.js:51]: Maybe
```suggestion
  allMinors: () => Object.keys(_versions),
```

File: docs/.vuepress/versions.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  /// All existing versions (across minor and patch)
+  allVersions: () => {
+    let out = [];
+    for (let key in _versions) {
+      out.push(..._versions[key].kuma);
+    }
+    return [...new Set(out)];
+  },
```
Comment (bartsmykla) [inline docs/.vuepress/versions.js:59]:
```suggestion
  allVersions: () => [...new Set(Object.values(_versions).flatMap(obj => obj.kuma))],
```

File: docs/.vuepress/versions.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  /// All minor versions that are not marked as "latest"
+  oldMinors: () => {
+    let out = [];
+    for (let key in _versions) {
+      if (!_versions[key].latest) {
+        out.push(key);
+      }
+    }
+    return out
+  }
```
Comment (bartsmykla) [inline docs/.vuepress/versions.js:77]:
```suggestion
  oldMinors: Object.keys(_versions).reduce((acc, curr) => {
    if (!_versions[curr].latest) {
      acc.push(curr);
    }
    
    return acc;
  }, []),
```

**Review by bartsmykla — APPROVED** [review]
Some suggestions, but overall lgtm

File: docs/docs/dev/installation/openshift.md (hunk 1 above)
Comment (michaelbeaumont) [reply docs/docs/dev/installation/openshift.md:29]: better than out of date links I suppose

File: docs/docs/dev/versions.json (hunk 2 above)
Comment (michaelbeaumont) [reply docs/docs/dev/versions.json:3]: OK, but it doesn't actually show up anywhere if I understood correctly?

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**